release: 0.9.0 (#3134)
rome/tools: CHANGELOG.md

# Rome changelog

## 0.9.0

### CLI

- You can now format content from standard input when using the command `rome format`:
```shell
echo "function f() { return {} }" | rome format --stdin-file-path example.js
```
the argument  `--stdin-file-path` is mandatory when formatting from standard in. The path should represent a
file name with its extension.
- Added `--apply-suggested` argument to the `rome check` command, to apply suggested and safe fixes.

> Suggested fixes should be considered **not stable** and applied with care. 

Reviewer: ```suggestion
Suggested fixes should be considered **unstable** and applied with care. 
```

Why so? It seems a bit confusing that the **recommended** rules are **unstable**. Why are they recommended?

Do we consider all code fixes unstable? If so, I think it would be easier to understand if this is mentioned as its own bullet point. 

* Code fixes are under development and are still unstable. But feel free to open an issue if you run into any problems

Reviewer 2: We could use a different wording here as "unstable" may seem related to the current status of the Rome linter, when this is an advice that applies generally to all code fixes: "Safe fixes" are code actions we're absolutely certain will not cause any syntax error or introduce changes in the behavior of the code, but due to the complex semantics of JavaScript most of code actions we emit are "Suggested fixes" that do not guarantee the same strict level of correctness.
Over time we'll certainly improve the existing actions and we may be able to raise the applicability level for some of them, but I think we'll always have actions we're not entirely certain about that the end user will have to either review manually, or explicitly allow with a flag like `--apply-suggested` if they know what they're doing.

Author: The concept of suggested fixes and safe fixes is not related to the concept of recommendation of the rules. A rule can be recommended but its fix may be suggested. A rule can be not recommended by us but its fix might be safe.  

Some fixes follows under the category of "suggested" due to the nature of JavaScript as dynamic language, and they are not bullet proof.

* Doc comments [#3129](https://github.com/rome/tools/pull/3129)

### Linter

- Changed the default severity for recommended rules to "error". You can [change the severity in the rome.json](https://rome.tools/#configure-a-rule).
- Added [`js/noExtraBooleanCast`](https://rome.tools/docs/lint/rules/noExtraBooleanCast/) lint rule.
- Added [`js/noDupeArgs`](https://rome.tools/docs/lint/rules/noDupeArgs/) lint rule.
- Added [`js/noShadowRestrictedNames`](https://rome.tools/docs/lint/rules/noShadowRestrictedNames/) lint rule.
- Added `js/inlineVariable` code action.
- Applied various stability fixes to the rule [`js/noUnusedVariables`](https://rome.tools/docs/lint/rules/noUnusedVariables/). [#3124](https://github.com/rome/tools/pull/3124) [#3060](https://github.com/rome/tools/pull/3060) [#3004](https://github.com/rome/tools/pull/3004)
- Fixed how the suggestion is applied [`js/noNegationElse`](https://rome.tools/docs/lint/rules/noNegationElse/). [#2999](https://github.com/rome/tools/issues/2999)

> - Fixed false positive in the rule [`js/noShoutyConstants`](https://rome.tools/docs/lint/rules/noShoutyConstants/). [#3077](https://github.com/rome/tools/issues/3077)
> - Fixed false positive in the rule [`ts/useShorthandArrayType`](https://rome.tools/docs/lint/rules/useShorthandArrayType/). [#3111](https://github.com/rome/tools/issues/3111)

Reviewer: ```suggestion
- Fixed a false positive in the rule [`js/noShoutyConstants`](https://rome.tools/docs/lint/rules/noShoutyConstants/). [#3077](https://github.com/rome/tools/issues/3077)
- Fixed a false positive in the rule [`ts/useShorthandArrayType`](https://rome.tools/docs/lint/rules/useShorthandArrayType/). [#3111](https://github.com/rome/tools/issues/3111)
```

# Rome changelog

## 0.9.0

### CLI

> - Added `--stdin-file-path` argument, needed when formatting from standard input.

Reviewer: As a user, it's unclear to me why this feature or functionality is necessary. Seems rather a technical detail. 

I assume it is for us to know the working directory and infer the file extension? I would then phrase it around that it is now possible to format files passed from stdin with the example you added further down.

- You can now format content from standard input when using the command `rome format`:
```shell
echo "function f() { return {} }" | rome format --stdin-file-path example.js
```
- Added `--apply-suggested` argument to the `rome check` command, to apply suggested and safe fixes.

### Linter

> - Now all rules will emit an error, check the website if you want to [change the severity of a rule](https://rome.tools/#configure-a-rule).

Reviewer: Where some rules only emiting warnings before? Or did we add the functionality to change the severity?

Author: Before ALL rules were emitting warning. Maybe  I will rephrase it, because only the recommended rules emit errors

Reviewer: > Changed the default severity for recommended rules to "error". You can change the severity in the `rome.json`

- Added `js/noExtraBooleanCast` lint rule.
- Added `js/noDupeArgs` lint rule.
- Added `js/noShadowRestrictedNames` lint rule.

> - Added `js/inlineVariable` code action.

Reviewer: Let's add links to the rules

- Fixed issues in the rule `js/noUnusedVariables`. [#3124](https://github.com/rome/tools/pull/3124) [#3060](https://github.com/rome/tools/pull/3060) [#3004](https://github.com/rome/tools/pull/3004)
- Fixed issues in the rule `js/noNegationElse`. [#2999](https://github.com/rome/tools/issues/2999)

> - Fixed issues in the rule `js/noShoutyConstants`. [#3077](https://github.com/rome/tools/issues/3077)

Reviewer: Can we be more specific `Fixed false positives in the rule ...`. Or what exactly was fixed

### Formatter

Greatly increased the compatibility with Prettier, here's an highlight of what we improved

##### `throw` and `return` arguments

##### Templates

##### Arrow functions

##### Conditional expressions

##### Parenthesis

##### Arrays

> ##### Loops

Reviewer: Do you want me to go in that much detail? I considered just to just enumerate the syntaxes where we made significant improvement

Author: I leave it to you. I just created headings based on the PRs closed in this release cycle.

### Configuration

> - **BREAKING CHANGE**: removed the second `"rules"` field from a field group.

Reviewer: Should this be at the top because it is a breaking change?

Author: I think it's fine. We have never been strict to "where" the breaking changes are, they are always listed as first thing under their respective area

  ```
- Fixed some edge cases where the comment suppressions were not working as expected.

### Linter

The linter is now marked as "alpha" and it can be used to lint code from the CLI and

> from the LSP.

Reviewer: How about moving the section about the unstable code fixes to here?

Author: The change is part of the previous release. I just moved its headings to have the same order in each release

### VSCode

> - **BREAKING CHANGE**: Removed the majority of settings that were available in the extension, use the

Reviewer: ```suggestion
- **BREAKING CHANGE**: Removed the majority of settings that were available in the extension, in favor of the
```
